torch/nn/attention/templated_attention.py

```diff
@@ -0,0 +1,81 @@
+"""This module implements the user facing API for templated attention in PyTorch."""
+import functools
+from typing import Callable
+
+import torch
+from torch._higher_order_ops.templated_attention import (
+    templated_attention as templated_attention_hop,
+)
+
+
+def _compose(*fs):
```

Comment:
btw, I think this is kind of a good example of the type of abstraction that's nice to write but quite annoying to handle without a generic program capture mechanism.